presto-main/src/main/java/com/facebook/presto/type/TypeRegistry.java

```diff
@@ -187,9 +188,18 @@ private Type instantiateParametricType(TypeSignature signature)
             parameters.add(typeParameter);
         }
 
-        ParametricType parametricType = parametricTypes.get(signature.getBase().toLowerCase(Locale.ENGLISH));
+        String baseSignature = signature.getBase().toLowerCase(Locale.ENGLISH);
```

Comment:
This logic should be moved to `FunctionAndTypeManager`. The whole reason for the refactoring is so logic related to `FunctionNamespaceManager` is all in one place.

Reply:
Maybe give me a few more days to put these logic in the right place if it's not quite clear what I mean yet.

Reply:
@rongrong I refaactored it a little, does this look more like what you had in mind now?